.ibm/pipelines/openshift-tests.sh

```diff
@@ -153,13 +153,34 @@ run_tests() {
 }
 
 check_backstage_running() {
-  # Check if Backstage is up and running
-  BACKSTAGE_URL_RESPONSE=$(curl -Is "https://${RELEASE_NAME}-${NAME_SPACE}.${K8S_CLUSTER_ROUTER_BASE}" | head -n 1)
+  local url="https://${RELEASE_NAME}-backstage-${NAME_SPACE}.${K8S_CLUSTER_ROUTER_BASE}"
+  
+  # Maximum number of attempts to check URL
+  local max_attempts=30
+  # Time in seconds to wait
+  local wait_seconds=30
+
+  echo "Checking if Backstage is up and running at $url"
+
+  for ((i=1; i<=max_attempts; i++)); do
+    # Get the status code
+    local http_status=$(curl -o /dev/null -s -w "%{http_code}\n" "$url")
+
+    # Check if the status code is 200
+    if [[ $http_status -eq 200 ]]; then
+      echo "Backstage is up and running!"
+      export BASE_URL=$url
```

Comment:
you don't need to export this if you're only echoing it to the console.

Reply:
We are using this in our e2e-tests